### PR TITLE
feat(api): record pvc usage in list executioncontexts api

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

Record pvc usage in list executioncontexts api

List executioncontexts return body:
```
{
    "metadata": {
        "total": 1,
        "itemsLength": 1
    },
    "items": [
        {
            "spec": {
                "cluster": "compass-stack",
                "namespace": "cyclone--devops",
                "pvc": "cyclone-pvc-devops"
            },
            "status": {
                "phase": "Ready",
                "reservedResources": {
                    "limits.cpu": "100m",
                    "limits.memory": "64Mi",
                    "requests.cpu": "50m",
                    "requests.memory": "32Mi"
                },
                "pvc": {
                    "usage": {
                        "total": "10.0G",
                        "used": "0",
                        "items": {
                            "workflowruns": "4.0K"
                        },
                        "usedPercentage": 0
                    },
                    "status": {
                        "phase": "Bound",
                        "accessModes": [
                            "ReadWriteMany"
                        ],
                        "capacity": {
                            "storage": "10Gi"
                        }
                    }
                }
            }
        }
    ]
}
```
**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @your-reviewer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by zhujian7** [comment]
@supereagle ptal

File: pkg/server/apis/v1alpha1/types.go [hunk 1]
```diff
@@ -265,7 +265,29 @@ type ExecutionContextStatus struct {
 	// can not be used by workflows execution.
 	ReservedResources map[core_v1.ResourceName]string `json:"reservedResources"`
 	// PVC describes status of PVC
-	PVC *core_v1.PersistentVolumeClaimStatus `json:"pvc"`
+	PVC PVCStatus `json:"pvc"`
+}
+
+// PVCStatus includes upstream kubernetes PersistentVolumeClaimStatus and human readable pvc usage profile
+type PVCStatus struct {
```
Comment (supereagle) [inline pkg/server/apis/v1alpha1/types.go]:
```suggestion
type PVCOverallStatus struct {
```

File: pkg/server/handler/v1alpha1/executioncontext.go [hunk 2]
```diff
@@ -71,6 +72,25 @@ func ListExecutionContexts(ctx context.Context, tenant string) (*types.ListRespo
 	return types.NewListResponse(len(executionContexts), executionContexts), nil
 }
 
+func getFullPVCStatus(tenant, pvcName string, cluster *api.ClusterSource) (fullStatus api.PVCStatus, err error) {
```
Comment (supereagle) [inline pkg/server/handler/v1alpha1/executioncontext.go]:
```suggestion
func getPVCOverallStatus(tenant, pvcName string, cluster *api.ClusterSource) (fullStatus api.PVCOverallStatus, err error) {
```

File: pkg/server/handler/v1alpha1/executioncontext.go [hunk 3]
```diff
@@ -71,6 +72,25 @@ func ListExecutionContexts(ctx context.Context, tenant string) (*types.ListRespo
 	return types.NewListResponse(len(executionContexts), executionContexts), nil
 }
 
+func getPVCOverallStatus(tenant, pvcName string, cluster *api.ClusterSource) (fullStatus api.PVCOverallStatus, err error) {
```
Comment (supereagle) [inline pkg/server/handler/v1alpha1/executioncontext.go]:
```suggestion
func getPVCOverallStatus(tenant, pvcName string, cluster *api.ClusterSource) (overallStatus api.PVCOverallStatus, err error) {
```

**Comment by zhujian7** [comment]
@supereagle PTAL

**Pull request merged**